### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by andriy-shymkiv**

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,34 @@
...
+    switch (obj.type) {
+      case 'addProperties':
+        Object.assign(copyState, obj.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of obj.keysToRemove) {
+          delete copyState[key];
+        }
+        break;
```
Comment (Oleksandr-Stryzhavchuk) [inline src/transformStateWithClones.js:23]: as I know, here needed a blank line before 'break'

**Review by Oleksandr-Stryzhavchuk — APPROVED** [review]
Good job!

File: src/transformStateWithClones.js (hunk 1 above)
Comment (Andriy-Shymkiv) [reply src/transformStateWithClones.js:23]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch#syntax

File: src/index.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  <meta charset="UTF-8">
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>Document</title>
+  <script src="transformStateWithClones.js"></script>
+</head>
+<body>
+
+</body>
+</html>
```
Comment (mykola-koval) [inline src/index.html:13]: need this?

File: src/transformStateWithClones.js [hunk 3]
```diff
@@ -7,7 +7,34 @@
...
+      case 'clear':
+        for (const key in copyState) {
+          delete copyState[key];
+        }
+        break;
+
+      default:
+        break;
+    }
+    newArray.push({ ...copyState });
```
Comment (mykola-koval) [inline src/transformStateWithClones.js:34]:
```suggestion
    }
    
    newArray.push({ ...copyState });
```

File: src/transformStateWithClones.js [hunk 4]
```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newArray = [];
+  const copyState = { ...state };
+
+  for (const obj of actions) {
```
Comment (mykola-koval) [inline src/transformStateWithClones.js:13]:
```suggestion
  for (const action of actions) {
```

**Review by mykola-koval — APPROVED** [review]

**Pull request closed without merging**